### ISSUE TITLE
New Truck Ticket with VIN# and VIN Images

### ISSUE THREAD

**Issue by kgilliam3662** [ConceptualSystems/sms-mobile-portable#13]
Installed SMSMobile 1.52, linked to Dropbox account successfully.
Click New Truck Ticket,  entered 145 as Driverid,  Picked Product Cars,  Click Enter VIN and entered
a VIN# and Decoded successfully.  click blue color, click ok.
click VIN# entry, choose Create Image,  click Size button and choose 1920x1080 since 1024x768 was not available.  click camera button to capture a picture.  Click VIN# entry, choose create image,
click snapshot button to save images, repeated 1 more time.  Now I'm back at New Truck Ticket.  I clicked Transmit and it seems to working uploading images to dropbox, but then it returned to the New Truck Ticket screen with all the information filled out.  I was expecting the screen to disappear back to the main menu screen like it normally does.  I click Transmit Again and it was much quicker, but returned to the main menu screen.  I then review my SMSMDS trace log file and only found the following:

```
04/30/2019 16:13:49 - <---: ConnID:[1] [{"STX":1,"TYPE":"REQUEST","MESSAGEID":"NEWTRUCKTICKET","ACTIVATIONKEY":"KBERTA","CUSTOMERID":"CASH","DRIVERID":"145","PRODUCTID":"CARS","TOWMAKEMODEL":"","TOWLICESENUMBER":"","SCALEID":1,"WEIGHT":0,"MANUAL":false,"DIRECTION":"I","VINNUMBER1":"1N6AA0ED4AN318993","VINMAKEMODEL1":"NISSAN TITAN 2010 N BLUE ","USERID":"ADMIN","ETX":1}]
04/30/2019 16:13:50 - GetScaleWeight: Scale#: 1, Weight: 5840, Motion: False, ScaleError: False
04/30/2019 16:13:52 - SAVING TICKET DATA ON YARD! TKT#: 301381
04/30/2019 16:13:52 - --->: ConnID:[1] [{"STX":1,"TYPE":"RESPONSE","MESSAGEID":"NEWTRUCKTICKET","MESSAGE":"Ticket Data Transmitted!","ETX":1}]
```

```
it's missing the "VINIMAGE1":"20190430_161159_vin1_drv145.jpg,20190430_161232_vin1_drv145.jpg,20190430_161242_vin1_drv145.jpg"

missing key/values pairs in the message.
```
I then checked the dropbox account and found the 3 files had been uploaded, but the message
seemed to be missing the files names you uploaded, or this message was received when I had to click the transmit button again and the images were already uploaded and you deleted the images name and or files on the tablet.   ????

**Comment by kgilliam3662** [comment]
```
JSON Message format:

Rem (NEW TRUCK TICKET) INBOUND TICKET Message TRANSMIT/REQUEST
{
'STX':1,
'TYPE':'REQUEST',
'MESSAGEID':'NEWTRUCKTICKET',
'ACTIVATIONKEY':'5497',
'CUSTOMERID':'TEST',
'DRIVERID':'9999',
'PRODUCTID':'CARS',
'TOWMAKEMODEL':'FORD F150 1999 N BLUE',
'TOWLICENSENUMBER':'324WXS TN',
'SCALEID':1,
'WEIGHT':0,
'MANUAL':false,
'DIRECTION':'I',
'VINNUMBER1':'JH4TB2H26CC000000',
'VINNUMBER2':'',
'VINNUMBER3':'',
'VINNUMBER4':'',
'VINNUMBER5':'',
'VINNUMBER6':'',
'VINNUMBER7':'',
'VINNUMBER8':'',
'VINMAKEMODEL1':'NISSAN SENTRA 1992 N RED',
'VINMAKEMODEL2':'',
'VINMAKEMODEL3':'',
'VINMAKEMODEL4':'',
'VINMAKEMODEL5':'',
'VINMAKEMODEL6':'',
'VINMAKEMODEL7':'',
'VINMAKEMODEL8':'',
'VINIMAGE1':'20180305_183921_TABLET_23423.JPG,,', 			'yyyymmdd_hhmmss_vin<VIN# 1-8>_drv<DRIVERID>.jpg (file uploaded to dropbox or tomcat!)
'VINIMAGE2':'',
'VINIMAGE3':'',
'VINIMAGE4':'',
'VINIMAGE5':'',
'VINIMAGE6':'',
'VINIMAGE7':'',
'VINIMAGE8':'',
'USERID':'CSCTECH',
'ETX':1
}
```

**Comment by moonlightcheese** [comment]
Do we need to send these empty lines like that?  And the excessive commas?

By the way, it appears that I am indeed sending the USERID along with this message.  Not sure if it's sent with floor scale as well or review ticket or other messages.
```
{
"STX":1,
"TYPE":"REQUEST",
"MESSAGEID":"NEWTRUCKTICKET",
"ACTIVATIONKEY":"KBERTA",
"CUSTOMERID":"CASH",
"DRIVERID":"152",
"PRODUCTID":"CARS",
"TOWMAKEMODEL":"CHEVROLET 1500 BLACK",
"TOWLICENSENUMBER":"QQQ666",
"SCALEID":1,
"WEIGHT":0,
"MANUAL":false,
"DIRECTION":"I",
"VINNUMBER1":"JT3AC12R8P1014055",
"VINNUMBER2":"NONE",
"VINMAKEMODEL1":"TOYOTA PREVIA 1993 N BLACK ",
"VINMAKEMODEL2":"ACURA CL N BLACK ",
"VINIMAGE1":"20190501_095243_vin1_drv152.jpg,20190501_095251_vin1_drv152.jpg,20190501_095258_vin1_drv152.jpg",
"VINIMAGE2":"20190501_095311_vin2_drv152.jpg",
"USERID":"ADMIN",
"ETX":1
}
```

**Comment by moonlightcheese** [comment]
Marking this as fixed for now.  If there is some issue with the extra empty data not being present, let me know.

Fixed in SMS Mobile 1.53.

**Comment by kgilliam3662** [comment]
without the empty lines, i have to check for null on every key/value pair.  I use the commas to help split
the 3 images into an array.  again, if you can't send them, then I will add code to check the array bounds to verify the count of images you sent me.

**Comment by moonlightcheese** [comment]
Okay I will add the empty lines.  In your example above, you did not have two commas on each of the VINIMAGE lines, which I assume is how you want it?

**Comment by moonlightcheese** [comment]
This is fixed in SMS Mobile 1.53.
